azure-mgmt-sql/src/main/java/com/azure/management/sql/implementation/SqlDatabaseImpl.java

```diff
@@ -0,0 +1,932 @@
+        PagedIterable<MetricInner> metricInners = this.sqlServerManager.inner().databases()
+            .listMetrics(this.resourceGroupName, this.sqlServerName, this.name(), filter);
+        if (metricInners != null) {
+            for (MetricInner metricInner : metricInners) {
+                sqlDatabaseMetrics.add(new SqlDatabaseMetricImpl(metricInner));
+            }
+        }
+        return Collections.unmodifiableList(sqlDatabaseMetrics);
+    }
+
+    @Override
+    public PagedFlux<SqlDatabaseMetric> listMetricsAsync(final String filter) {
+        return this.sqlServerManager.inner().databases()
+            .listMetricsAsync(this.resourceGroupName, this.sqlServerName, this.name(), filter)
+            .mapPage(metricInner -> new SqlDatabaseMetricImpl(metricInner));
+    }
+
+    @Override
+    public List<SqlDatabaseMetricDefinition> listMetricDefinitions() {
+        List<SqlDatabaseMetricDefinition> sqlDatabaseMetricDefinitions = new ArrayList<>();
+        PagedIterable<MetricDefinitionInner> metricDefinitionInners = this.sqlServerManager.inner().databases()
+            .listMetricDefinitions(this.resourceGroupName, this.sqlServerName, this.name());
+        if (metricDefinitionInners != null) {
+            for (MetricDefinitionInner metricDefinitionInner : metricDefinitionInners) {
+                sqlDatabaseMetricDefinitions.add(new SqlDatabaseMetricDefinitionImpl(metricDefinitionInner));
+            }
+        }
+
+        return Collections.unmodifiableList(sqlDatabaseMetricDefinitions);
+    }
+
+    @Override
+    public PagedFlux<SqlDatabaseMetricDefinition> listMetricDefinitionsAsync() {
+        return this.sqlServerManager.inner().databases()
+            .listMetricDefinitionsAsync(this.resourceGroupName, this.sqlServerName, this.name())
+            .mapPage(metricDefinitionInner -> new SqlDatabaseMetricDefinitionImpl(metricDefinitionInner));
+    }
+
+    @Override
+    public TransparentDataEncryption getTransparentDataEncryption() {
+        TransparentDataEncryptionInner transparentDataEncryptionInner = this.sqlServerManager.inner()
+            .transparentDataEncryptions().get(this.resourceGroupName, this.sqlServerName, this.name());
+        return (transparentDataEncryptionInner == null) ? null : new TransparentDataEncryptionImpl(this.resourceGroupName, this.sqlServerName, transparentDataEncryptionInner, this.sqlServerManager);
+    }
+
+    @Override
+    public Mono<TransparentDataEncryption> getTransparentDataEncryptionAsync() {
+        final SqlDatabaseImpl self = this;
+        return this.sqlServerManager.inner()
+            .transparentDataEncryptions().getAsync(this.resourceGroupName, this.sqlServerName, this.name())
+            .map(transparentDataEncryptionInner -> new TransparentDataEncryptionImpl(self.resourceGroupName, self.sqlServerName, transparentDataEncryptionInner, self.sqlServerManager));
+    }
+
+    @Override
+    public Map<String, ServiceTierAdvisor> listServiceTierAdvisors() {
```

Comment:
Just share sample to [convert PagedFlux to map](https://github.com/Azure/azure-libraries-for-java/blob/vnext/azure-mgmt-compute/src/main/java/com/azure/management/compute/implementation/VirtualMachineExtensionsImpl.java#L55-L60). This is optional.